skimage/feature/peak.py

```diff
@@ -92,6 +95,19 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
     array([[10, 10, 10]])
 
     """
+    def get_high_intensity_peaks(image, mask, num_peaks):
+        """
+        Return the highest intensity peak coordinates.
+        """
+        # get coordinates of peaks
+        coord = np.transpose(mask.nonzero())
+        # select num_peaks peaks
+        if coord.shape[0] > num_peaks:
+            intensities = image.flat[np.ravel_multi_index(coord.transpose(),
+                                                          image.shape)]
+            idx_maxsort = np.argsort(intensities)[::-1]
+            coord = coord[idx_maxsort][:num_peaks]
+        return coord
```

Comment:
Well, I just saw that my indexing is wrong, should be `[idx_maxsort][-num_peaks:]`, but other than that it should work...?